components/script/dom/bindings/buffer_source.rs

```diff
@@ -267,26 +288,27 @@ where
         }
     }
 
-    pub fn array_length(&self) -> usize {
+    pub(crate) fn array_length(&self) -> usize {
         self.get_buffer().unwrap().len()
     }
 
+    /// <https://tc39.es/ecma262/#typedarray>
     pub(crate) fn has_typed_array_name(&self) -> bool {
         match &self.buffer_source {
-            BufferSource::Int8Array(_)
-            | BufferSource::Int16Array(_)
-            | BufferSource::Int32Array(_)
-            | BufferSource::Uint8Array(_)
-            | BufferSource::Uint16Array(_)
-            | BufferSource::Uint32Array(_)
-            | BufferSource::Uint8ClampedArray(_)
-            | BufferSource::BigInt64Array(_)
-            | BufferSource::BigUint64Array(_)
-            | BufferSource::Float32Array(_)
-            | BufferSource::Float64Array(_)
-            | BufferSource::ArrayBuffer(_)
-            | BufferSource::Default(_) => true,
-            BufferSource::DataView(_) => false,
+            BufferSource::Int8Array(buffer) |
+            BufferSource::Int16Array(buffer) |
+            BufferSource::Int32Array(buffer) |
+            BufferSource::Uint8Array(buffer) |
+            BufferSource::Uint16Array(buffer) |
+            BufferSource::Uint32Array(buffer) |
+            BufferSource::Uint8ClampedArray(buffer) |
+            BufferSource::BigInt64Array(buffer) |
+            BufferSource::BigUint64Array(buffer) |
+            BufferSource::Float32Array(buffer) |
+            BufferSource::Float64Array(buffer) |
+            BufferSource::Default(buffer) |
+            BufferSource::ArrayBuffer(buffer) |
+            BufferSource::DataView(buffer) => unsafe { JS_IsTypedArrayObject(*buffer.handle()) },
```

Comment:
Here I would rather do like before(so not just `JS_` because from the type we should be able to tell), except that it should return false for `ArrayBuffer`.

Reply:
> what if someone miss use the API

Then I think we should check this when it is initialized.

But, if as suggested elsewhere, we reduce the enum to two variants, then for the view one we would have to use this API to differentiate between a data view and a typed array. And I think that's something we should do actually.